### DISCUSSION TITLE
[A-L] (14 marzo 2024) Esercizio 3

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#19] [Esercitazioni A-L]
Durante una visita in profondità, ad ogni vertice $u$ di un grafo viene associato un colore che varia nel tempo $t$ dell’algoritmo.

Se $t < t[u]$ e cioè il vertice $u$ deve ancora essere stato visitato per la prima volta allora u è colorato di $BIANCO$. 
Se $t[u] ≤ t ≤ T[u]$ e cioè il vertice $u$ è ancora in visita allora u è colorato di $GRIGIO$. 
Infine, se $t > T[u]$ e cioè il vertice $u$ è stato già completamente visitato allora u è colorato di $NERO$. 

Fare una tabella 3x3 con righe e colonne contrassegnate da $BIANCO$, $GRIGIO$ e $NERO$. In ogni cella $(i, j)$, indicare se, in un alcun punto di una ricerca in profondità di un grafo diretto ,è possibile avere un arco da un vertice di colore $i$ ad un vertice di colore $j$. Per ogni possibile arco, indicare se esso può essere:

- arco dell’albero,
-  arco all’indietro,
-  arco in avanti o
-  arco di attraversamento.

Fare una seconda tabella per una ricerca in profondità di un grafo non diretto.

![image](https://github.com/sapienzastudentsnetwork/progalg2324/assets/40034429/e6709224-5834-475f-8024-9e1221e4f45e)

**Comment by Frisbon** [comment]
Molto probabilmente sbagliato

Non so se la classificazione degli archi varia a seconda della struttura del grafo...
![image_2024-06-14_10-49-07](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/cff1df58-51c1-43a4-aea5-9392a7bb1b70)
